cpp/include/cudf/rolling.hpp

```diff
@@ -379,6 +381,119 @@ std::unique_ptr<column> grouped_time_range_rolling_window(
   std::unique_ptr<aggregation> const& aggr,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief  Applies a grouping-aware, value range-based rolling window function to the values in a
+ *         column.
+ *
+ * This is a generalization of `grouped_time_range_rolling_window()` to work with non-timestamp
+ * columns. Like `grouped_time_range_rolling_window()`, this function aggregates values in a window
+ * around each element of a specified `input` column. It differs from
+ * `grouped_time_range_rolling_window()` in that instead of a timestamp column, the ordered column
+ * can be of integral types as well as timestamps.
+ *
+ *   1. The elements of the `input` column are grouped into distinct groups (e.g. the result of a
+ *      groupby), determined by the corresponding values of the columns under `group_keys`. The
+ *      window-aggregation cannot cross the group boundaries.
+ *   2. Within a group, the aggregation window is calculated based on an interval (e.g. number
+ *      of days preceding/following the current row). The value intervals are applied on the
+ *      `orderby_column` argument.
+ *
+ * Note: This method requires that the rows are presorted by the group keys and orderby column
+ * values.
+ *
+ * The window intervals are specified as scalar values appropriate for the orderby column:
+ *   1. If `orderby` column is a timestamp, the `preceding`/`following` windows are specified
+ *      in terms of lower resolution `duration` scalars.
+ *      E.g. For `orderby` column of type `TIMESTAMP_SECONDS`, the intervals may be
+ * `DURATION_SECONDS` or `DURATION_DAYS`. Higher resolution durations (e.g. `DURATION_NANOSECONDS`)
+ * cannot be used with lower resolution timestamps.
+ *   2. If the `orderby` column is an integral type (e.g. `INT32`), the `preceding`/`following`
+ *      should be the exact same type (`INT32`).
+ *
+ * @code{.pseudo}
+ * Example: Consider a user-sales dataset, where the rows look as follows:
+ *  { "user_id", sales_amt, date }
```

Comment:
The whole point of this API is to support orderby columns that aren't timestamps, yes? I think an example that doesn't use dates would be more helpful.

Reply:
Two examples, maybe. One with dates and duration intervals. Another with a numeric range.

Reply:
Two examples would make it too long to read. I've added a contrived example that doesn't use dates.